internal/dataplane/kongstate/kongstate.go

```diff
@@ -74,7 +75,7 @@ func (ks *KongState) FillConsumersAndCredentials(log logrus.FieldLogger, s store
 				"secret_name":      cred,
 				"secret_namespace": consumer.Namespace,
 			})
-			secret, err := s.GetSecret(consumer.Namespace, cred)
+			secret, err := s.GetSecret(context.TODO(), consumer.Namespace, cred)
```

Comment:
We'll we actually get to replacing all those TODO contexts at some point 😅 ?

Reply:
These TODOs were actually an artifact of using the `kongstate.SecretGetter` interface in the validator (which is implemented by the `store.Store` and it makes sense for it to not accept context as it does operate on local caches). I've added a separate `SecretGetter` interface in the validator package: f626fa0

Reply:
I'm confused. What do we need 2? Because one doesn't utilize `context.Context` (because it's local 🤔 ? How does that change things?)

Reply:
Before this PR's changes, validator was using a thin wrapper around a manager client `managerClientSecretGetter` which did not accept a context: https://github.com/Kong/kubernetes-ingress-controller/blob/85133ff040d93348c21a2690746b6dc5a223d02f/internal/admission/validator.go#L406. It should accept one though because the manager client `Get` method accepts it. 

I modified the `managerClientSecretGetter` wrapper to properly accept a context that made it not compliant with the `kongstate.SecretGetter` interface used before. Because of that, I had to introduce the interface in the validator package.